core/data.go

```diff
@@ -554,61 +562,57 @@ func (pm *PaymentMetadata) UnmarshalDynamoDBAttributeValue(av types.AttributeVal
 		return fmt.Errorf("expected *types.AttributeValueMemberM, got %T", av)
 	}
 	pm.AccountID = m.Value["AccountID"].(*types.AttributeValueMemberS).Value
-	binIndex, err := strconv.ParseUint(m.Value["BinIndex"].(*types.AttributeValueMemberN).Value, 10, 32)
+	reservationPeriod, err := strconv.ParseUint(m.Value["ReservationPeriod"].(*types.AttributeValueMemberN).Value, 10, 32)
 	if err != nil {
-		return fmt.Errorf("failed to parse BinIndex: %w", err)
+		return fmt.Errorf("failed to parse ReservationPeriod: %w", err)
 	}
-	pm.BinIndex = uint32(binIndex)
+	pm.ReservationPeriod = uint32(reservationPeriod)
 	pm.CumulativePayment, _ = new(big.Int).SetString(m.Value["CumulativePayment"].(*types.AttributeValueMemberN).Value, 10)
 	return nil
 }
 
 func (pm *PaymentMetadata) ToProtobuf() *commonpb.PaymentHeader {
 	return &commonpb.PaymentHeader{
 		AccountId:         pm.AccountID,
-		BinIndex:          pm.BinIndex,
+		ReservationPeriod: pm.ReservationPeriod,
 		CumulativePayment: pm.CumulativePayment.Bytes(),
+		Salt:              pm.Salt,
 	}
 }
 
 // ConvertPaymentHeader converts a protobuf payment header to a PaymentMetadata
 func ConvertPaymentHeader(header *commonpb.PaymentHeader) *PaymentMetadata {
 	return &PaymentMetadata{
 		AccountID:         header.AccountId,
-		BinIndex:          header.BinIndex,
+		ReservationPeriod: header.ReservationPeriod,
 		CumulativePayment: new(big.Int).SetBytes(header.CumulativePayment),
+		Salt:              header.Salt,
 	}
 }
 
 // ConvertToProtoPaymentHeader converts a PaymentMetadata to a protobuf payment header
 func (pm *PaymentMetadata) ConvertToProtoPaymentHeader() *commonpb.PaymentHeader {
 	return &commonpb.PaymentHeader{
 		AccountId:         pm.AccountID,
-		BinIndex:          pm.BinIndex,
+		ReservationPeriod: pm.ReservationPeriod,
 		CumulativePayment: pm.CumulativePayment.Bytes(),
+		Salt:              pm.Salt,
 	}
 }
 
 // ConvertToProtoPaymentHeader converts a PaymentMetadata to a protobuf payment header
 func ConvertToPaymentMetadata(ph *commonpb.PaymentHeader) *PaymentMetadata {
 	return &PaymentMetadata{
 		AccountID:         ph.AccountId,
-		BinIndex:          ph.BinIndex,
+		ReservationPeriod: ph.ReservationPeriod,
 		CumulativePayment: new(big.Int).SetBytes(ph.CumulativePayment),
+		Salt:              ph.Salt,
 	}
 }
```

Comment:
There are two sets of methods to convert from/to protobuf (there are `ToProtobuf` & `ConvertPaymentHeader` above). Let's just keep 1